ols/src/query_helpers/query_docs.py

```diff
@@ -0,0 +1,124 @@
+"""This module has the components to retrieve docs from vector store."""
+from typing import Any, List
+
+from langchain_core.documents.base import Document
+from langchain_core.vectorstores import VectorStore, VectorStoreRetriever
+
+from ols.utils.logger import Logger
+
+
+class QueryDocs:
+    """Generic interface for retrieving docs from any vector store."""
+
+    def __init__(self) -> None:
+        """Initialize the `QueryDocs` instance."""
+        self.logger = Logger("QueryDocs").logger
+
+    def get_relevant_docs(
+        self,
+        query: str,
+        perform_query: bool = True,
+        vectordb: VectorStore = None,
+        search_type: str = "similarity",
+        **search_kwargs: Any,
+    ) -> List[Document] | None:
```

Comment:
Instead of `List` from typing, use the built-in generic type `list`.

Reply:
@onmete The whole class is a wrapper around langchain code. The reason why some of the elements are what it is there is because of the way langchain code has been written. Please check the return type here https://github.com/langchain-ai/langchain/blob/master/libs/core/langchain_core/retrievers.py#L182

Reply:
That shouldn't matter. You can try with linter, but we are typing the return type of our code. Given aside the `List` and `list` is actually the same thing, since python 3.9 (give or take) it is recommended to use python build-ins, just because you don't need to import "the same" thing  from somewhere else. (it wasn't possible in earlier versions)

Reply:
As I mentioned this is a wrapper and data structures will follow close to what has the class is wrapping around.

Reply:
You are missing the point here.

Reply:
The langchain is not returning `List`, but `list`. They just use `List` in the typing. Because of reasons I explained above. Most likely because they are still supporting python 3.8, where you can't use buildins as type hints.

Reply:
`list` (lowercase) is perfectly fit there. You can check types - it will be ok

Reply:
done